src/core/time_arrays.jl

```diff
@@ -176,6 +215,41 @@ function Base.getindex(mat::TimestepMatrix{VariableTimestep{D_TIMES}, T, 2}, idx
 	_missing_data_check(data, t)
 end
 
+function Base.getindex(mat::TimestepMatrix{FixedTimestep{FIRST, STEP, LAST}, T, 1}, ts::TimestepValue{T}, idx::AnyIndex) where {T, FIRST, STEP, LAST} 
+	t = _get_time_value_position([FIRST:STEP:LAST...], ts)
+	data = mat.data[t, idx]
+	_missing_data_check(data, t)
+end
+
+function Base.getindex(mat::TimestepMatrix{VariableTimestep{TIMES}, T, 1}, ts::TimestepValue{T}, idx::AnyIndex) where {T, TIMES}
+	t = _get_time_value_position(TIMES, ts)
+	data = mat.data[t, idx]
+	_missing_data_check(data, t)
+end
+
+function Base.getindex(mat::TimestepMatrix, ts::TimestepIndex, idx::AnyIndex)
+	t = ts.index
+	size(mat.data, 1) < t ? error("TimestepIndex index $t extends beyond end of first dim of TimestepMatrix") : data = mat.data[t, idx]
```

Comment:
I would write a function for it as soon as I see it's needed in more than 1 place, though it depends how trivial the redundancy is. For something like this, that's enforcing a logical constraint and emitting an error message, I would definitely write a function rather than cut & paste.

Reply:
Yea, I can generalize the error message a little bit and then make a function!